### PR TITLE
Push Notification Refactor

### PR DESCRIPTION
**Pull request by lord-tx**
Refactor the push notification service class adding types for forward compatibility.

### CODE REVIEW

File: lib/model/parsed_push_notification.dart [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    }
+    if (data?['placeId'] != null && data?['placeId'] != '') {
+      return PushNotificationParsedType.place;
+    }
+    return PushNotificationParsedType.noPayload;
+  }
+
+  /// Simply trigger an action based off its [PushNotificationParsedType]
+  void actionToPerform() {
+    final _loggingService = getLogger("Parsed Push Notification");
```
Comment (asogwachiuba) [inline lib/model/parsed_push_notification.dart:53]: There's a logging service that is used throughout the codebase -logger.d("message to log")-. But if you used this one for some other reason then okay.

**Review by asogwachiuba — APPROVED** [review]

File: lib/model/parsed_push_notification.dart (hunk 1 above)
Comment (lord-tx) [reply lib/model/parsed_push_notification.dart:53]: Noted. 

Although both of these logging services already existed in the code base and based on the 'Stacked' framework, I thought we'd be using the Stacked generated one.

Comment (lord-tx) [reply lib/model/parsed_push_notification.dart:53]: I would switch to the other one to ensure consistency across logging.

Comment (asogwachiuba) [reply lib/model/parsed_push_notification.dart:53]: Yeah okay.

**Review by divincoder — APPROVED** [review]

**Pull request merged**